haskell/protobuf.bzl

```diff
@@ -201,12 +201,14 @@ _haskell_proto_aspect = aspect(
 )
 
 def _haskell_proto_library_impl(ctx):
-    dep = ctx.attr.deps[0]  # FIXME
-    return [
-        dep[HaskellBuildInfo],
-        dep[HaskellLibraryInfo],
-        DefaultInfo(files = dep[HaskellProtobufInfo].files),
-    ]
+    res = []
+    defaultInfoFilesDepset = depset()
+    for dep in ctx.attr.deps:
+        res.append(dep[HaskellBuildInfo])
+        res.append(dep[HaskellLibraryInfo])
+        defaultInfoFilesDepset = depset(transitive = [defaultInfoFilesDepset, dep[HaskellProtobufInfo].files])
```

Comment:
Also, I looked at some docs for `DefaultInfo` [0] which indicate that it has many fields, is it bad to only set `files`?

0: https://docs.bazel.build/versions/master/skylark/lib/DefaultInfo.html